books/RayTracingInOneWeekend.html

```diff
@@ -502,54 +502,94 @@
+  those rays.
+The involved steps are
 
-    1. Calculate the ray from the eye to the pixel,
+    1. Calculate the ray from the “eye” through the pixel,
     2. Determine which objects the ray intersects, and
     3. Compute a color for that intersection point.
 
 When first developing a ray tracer, I always do a simple camera for getting the code up and running.
 I also make a simple `ray_color(ray)` function that returns the color of the background (a simple
-gradient).
+  gradient).
 
 I’ve often gotten into trouble using square images for debugging because I transpose $x$ and $y$ too
-often, so I’ll use a non-square image.
+  often, so I’ll use a non-square image.
 For now we'll use a 16:9 aspect ratio, since that's so common.
+A 16:9 aspect ratio means that the image width over image height = 16/9.
+For example, as an image 800 pixels wide by 400 pixels high has a 2:1 aspect ratio.
+
+We will drive the image (and hence viewport) size from the desired aspect ratio and the image width.
+That way, you can increase or decrease your image size by changing a single value, instead of doing
+  a calculation and changing two values, which you'd have to do if you had to specify both the image
+  width and image height for some target aspect ratio.
+Also, in calculating the image height from the image width and the desired aspect ratio, we'll need
+  to ensure that the resulting height is at least 1.
 
 In addition to setting up the pixel dimensions for the rendered image, we also need to set up a
-virtual viewport through which to pass our scene rays. For the standard pixel spacing (equally
-spaced in both the horizontal and vertical direction), the viewport's aspect ratio should be the
-same as our rendered image. We'll just pick a viewport two units in height. We'll also set the
-distance between the projection plane and the projection point to be one unit. This is referred to
-as the _focal length_, not to be confused with the _focus distance_, which we'll present later.
+  virtual _viewport_ through which to pass our scene rays.
+The viewport is a rectangle in the 3D world that contains the grid of image pixel locations.
+For the standard pixel spacing (equally spaced in both the horizontal and vertical direction), the
+  viewport will have the same aspect ratio as the rendered image.
+
+To start things off, we'll arbitrarily choose a viewport height of 2.0, and scale the viewport width
+  to give us the desired aspect ratio.
+Here's a sketch of what this code will look like:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    auto aspect_ratio = 16.0 / 9.0;
+    int image_width = 400;
+
+    // Calculate the image height, and ensure that it's at least 1.
+    int image_height = static_cast<int>(image_width / aspect_ratio);
+    image_height = (image_height < 1) ? 1 : image_height;
+
+    auto viewport_height = 2.0;
```

Comment:
Consider adding a comment here.:

"// Calculate the viewport height, we don't need to check for 1 because it's a double.

Reply:
Did this:
```
    // Calculate the viewport height. Since it's real valued, we don't check for values less then 1.
```

Reply:
Sorry to nitpick. I prefer
```
// Calculate the viewport height. We don't check for values less than 1 since it's real valued
```

No passive voice in comments

Reply:
I don't see any passive voice. It's a combination of command and active. Are you referring to the predicate "since it's real valued"?

Anyway, latest iteration is `// Viewport widths less than one are ok since they are real valued.`

Reply:
I prefer  "We do Y because X" then "since X, we do Y"

Reply:
Ok. Well, looks like the above comment satisfies this.